client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts

```diff
@@ -26,10 +31,14 @@ export const isRefactoredForThankYouV2 = ( props: CheckoutThankYouCombinedProps
 		return true;
 	}
 
+	if ( isTitanWithoutMailboxes( props.selectedFeature ) && getDomainPurchase( purchases ) ) {
```

Comment:
Similarly to the earlier comment in `<CheckoutThankYou>` component. We probably don't need to get the domain purchase, instead, I would check for titan purchase.